### PR TITLE
Gangams/telegraf update to 1.28.5

### PR DESCRIPTION
**Pull request by ganga1980**
This PR has following updates
1.  Linux Agent, Telegraf update from 1.27.3 to 1.28.5
3.  update from response_timeout to timeout since response_timeout deprecated and new field name is timeout
4.  comment bearer_token since this is not required and also to address customer incident
5. re-enable timebound token

scenarios validaed
1.  Pod annotation scraping for both Linux (prom sidecar) & Windows
2.  Cluster level service endpoint scraping through Replica
3.  default metrics for both Linux and windows

Note: Telegraf upgrade for windows requires more investigation as the latest versions doesnt work.

### CODE REVIEW

**Comment by ganga1980** [comment]
ama-logs-rs memory usage(p99)
image
ama-logs-rs cpu usage(p99)
image
ama-logs ds memory usage (p99)
image
ama-logs ds cpu usage (p99)
image

File: build/windows/installer/conf/telegraf.conf [hunk 1]
```diff
@@ -124,7 +124,7 @@
 #Prometheus Custom Metrics
 [[inputs.prometheus]]
   interval = "$AZMON_TELEGRAF_CUSTOM_PROM_INTERVAL"
-  
+
```
Comment (jatakiajanvi12) [inline build/windows/installer/conf/telegraf.conf]: Unnecessary change in file

File: build/linux/installer/conf/telegraf-prom-side-car.conf [hunk 2]
```diff
@@ -184,12 +192,12 @@
   # kube_config = "/path/to/kubernetes.config"
 
   ## Use bearer token for authorization. ('bearer_token' takes priority)
-  bearer_token = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+  # bearer_token = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```
Comment (jatakiajanvi12) [inline build/linux/installer/conf/telegraf-prom-side-car.conf:195]: Should we just remove instead of commenting? Or does it get used somewhere?

File: build/linux/installer/conf/telegraf-prom-side-car.conf [hunk 3]
```diff
@@ -66,7 +66,15 @@
   ## Run telegraf in quiet mode (error log messages only).
   quiet = true
   ## Specify the log file name. The empty string means to log to stderr.
-  logfile = ""
+  logfile = "/var/opt/microsoft/docker-cimprov/log/telegraf_error.log"
+
+  ## Maximum number of rotated archives to keep, any older logs are deleted. If set to -1, no archives are removed.
+  logfile_rotation_max_archives = 3
```
Comment (jatakiajanvi12) [inline build/linux/installer/conf/telegraf-prom-side-car.conf:72]: Do we need 3 log files as archives?

File: build/linux/installer/conf/telegraf-prom-side-car.conf (hunk 2 above)
Comment (jatakiajanvi12) [reply build/linux/installer/conf/telegraf-prom-side-car.conf:195]: How was it affecting the customer?

File: charts/azuremonitor-containers/templates/ama-logs-daemonset-windows.yaml [hunk 4]
```diff
@@ -98,6 +98,9 @@ spec:
        - name: SIDECAR_SCRAPING_ENABLED
          value: {{ .Values.amalogs.sidecarscraping | quote }}
        volumeMounts:
+        - name: kube-api-access
+          mountPath: /var/run/secrets/kubernetes.io/serviceaccount
```
Comment (jatakiajanvi12) [inline charts/azuremonitor-containers/templates/ama-logs-daemonset-windows.yaml]: Does this timebound token work for the scenario of the sev2? Can you add that in the testing scenarios you mentioned as well?

**Comment by jatakiajanvi12** [comment]
> ama-logs-rs memory usage(p99) image ama-logs-rs cpu usage(p99) image ama-logs ds memory usage (p99) image ama-logs ds cpu usage (p99) image

Why is cpu increasing for RS?

File: build/windows/installer/conf/telegraf.conf (hunk 1 above)
Comment (ganga1980) [reply build/windows/installer/conf/telegraf.conf]: > Unnecessary change in file

Yes, this is number of log files to keep and 3 should be fine with each 1MB. other places we have 5 - https://github.com/microsoft/Docker-Provider/blob/ci_prod/kubernetes/linux/main.sh#L940.

File: build/linux/installer/conf/telegraf-prom-side-car.conf (hunk 3 above)
Comment (ganga1980) [reply build/linux/installer/conf/telegraf-prom-side-car.conf:72]: > Do we need 3 log files as archives?

Yes, this is number of log files to keep and 3 should be fine with each 1MB. other places we have 5 - https://github.com/microsoft/Docker-Provider/blob/ci_prod/kubernetes/linux/main.sh#L940.

**Comment by ganga1980** [comment]
> > ama-logs-rs memory usage(p99) image ama-logs-rs cpu usage(p99) image ama-logs ds memory usage (p99) image ama-logs ds cpu usage (p99) image
> 
> Why is cpu increasing for RS?

> the cpu increase is very little i.e. 25millis to 45millis and this can be ignored. This increase because of the telegraf.

File: charts/azuremonitor-containers/templates/ama-logs-daemonset-windows.yaml (hunk 4 above)
Comment (ganga1980) [reply charts/azuremonitor-containers/templates/ama-logs-daemonset-windows.yaml]: > Does this timebound token work for the scenario of the sev2? Can you add that in the testing scenarios you mentioned as well?

> With the updated telegraf has the fix for refreshing the token. Yes, this works.  There is no special testing scenario and just applying the yaml to this cluster should restrict the token expiry for an hour and all the components should refresh the token before the expiry.

File: build/linux/installer/conf/telegraf-prom-side-car.conf (hunk 2 above)
Comment (ganga1980) [reply build/linux/installer/conf/telegraf-prom-side-car.conf:195]: > How was it affecting the customer?

> customer app validating this token and this token shouldnt be sent as part of the request since all pod based scraping happens without any auth.

**Review by jatakiajanvi12 — DISMISSED** [review]

**Review by jatakiajanvi12 — APPROVED** [review]
